### PR TITLE
Command parameter + Command Source Code Command

### LINKED ISSUE #1171: Command Source Code Command
We should have a command that when ran, links to the GitHub page for that command's code. We'd have to somehow detect what branch or commit the bot is running on and link to the right place on GitHub.

## Example

If a user types `s!commandsource isitup`, the bot would respond with `https://github.com/SwitchbladeBot/switchblade/blob/master/src/commands/utility/isitup.js`

### PR DESCRIPTION
**Pull request by almeidx**
Closes #1171

### CODE REVIEW

File: src/utils/GitUtils.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+const axios = require('axios')
+const fs = require('fs')
+
+module.exports = class GitUtils {
+  static async getHashOrBranch () {
+    const rev = fs.readFileSync('.git/HEAD').toString().trim()
```
Comment (pedrofracassi) [inline src/utils/GitUtils.js:6]: What happens if `.git/HEAD` doesn't exist?

**Review by pedrofracassi — CHANGES_REQUESTED** [review]

File: src/utils/GitUtils.js (hunk 1 above)
Comment (almeidx) [reply src/utils/GitUtils.js:6]: The only reason that I see that could happen is if there was no local repository, and on that case the command just becomes obsolete

Comment (almeidx) [reply src/utils/GitUtils.js:6]: But anyway, I will take that into account

Comment (almeidx) [reply src/utils/GitUtils.js]: https://github.com/SwitchbladeBot/switchblade/pull/1173/commits/578af3c5aea34bd53728c394f81a8e9dd9b8a443

File: src/commands/bot/commandsource.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    }, client)
+  }
+
+  async run ({ channel, author, t }, command) {
+    const branchOrHash = await GitUtils.getHashOrBranch()
+    if (branchOrHash === null) throw new CommandError(t('commands:commandsource.noRepositoryOrHEAD'))
+    if (!branchOrHash) throw new CommandError(t('commands:commandsource.branchNotUpToDate'))
+
+    channel.send(new SwitchbladeEmbed(author)
+      .setTitle(command.name)
```
Comment (pedrofracassi) [inline src/commands/bot/commandsource.js:28]:
```suggestion
      .setTitle(command.fullName)
```

Discord_FMXIXNxvwF

File: src/commands/bot/commandsource.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+      .setDescription(`[${t('commands:commandsource.commandSource')}](${REPOSITORY_URL}/blob/${branchOrHash}/${command.path.split(sep).join('/')}) ([${branchOrHash}](${REPOSITORY_URL}/tree/${branchOrHash}))`)
```
Comment (pedrofracassi) [inline src/commands/bot/commandsource.js:29]: Maybe try something like this?

image

**Review by pedrofracassi — CHANGES_REQUESTED** [review]

File: src/commands/bot/commandsource.js (hunk 3 above)
Comment (almeidx) [reply src/commands/bot/commandsource.js]: https://github.com/SwitchbladeBot/switchblade/pull/1173/commits/545e5c4c7f356bdde46193c0eafbe65ea46f762c

**Review by pedrofracassi — CHANGES_REQUESTED** [review]
image

image

Dates don't seem to match

**Review by pedrofracassi — CHANGES_REQUESTED** [review]
Maybe make the bot type while it's fetching data from GitHub?

**Comment by almeidx** [comment]
> image
> 
> image
> 
> Dates don't seem to match

Cannot replicate?
https://i.imgur.com/Hh56Z5Q.png
https://i.imgur.com/gq0Db5G.png

**Review by Doges — COMMENTED** [review]
LGTM

**Review by Doges — DISMISSED** [review]
LGTM

**Review by pedrofracassi — DISMISSED** [review]
LGTM I guess

**Review by jaxxibae — CHANGES_REQUESTED** [review]
image
image

**Comment by almeidx** [comment]
@jaxxibae The "using master as base" only applies to the URLs, not the "Last edited by" part

**Comment by jaxxibae** [comment]
Oh? My bad then

**Review by jaxxibae — APPROVED** [review]
LGTM

**Comment by jaxxibae** [comment]
PS: resolve conflicts @almeidx

**Comment by almeidx** [comment]
> PS: resolve conflicts @almeidx

Done

**Review by Doges — APPROVED** [review]
LGTM

**Review by metehus — APPROVED** [review]
LGTM

**Pull request merged**